### PR TITLE
Bug 1882490: data/azure/master: Add dash to nic name

### PR DESCRIPTION
**Pull request by mtnbikenc**
Fix code consistency issue of missing dash in nic name.

```
> $ git grep '${var.cluster_id}-master-${count.index}' | wc -l
21

> $ git grep '${var.cluster_id}-master${count.index}' | wc -l 
1
```

### CODE REVIEW

**Comment by mjudeikis** [comment]
Appreciate the change but before we go this path I want to understand how this is aligned with other teams involved: 

1. What is the plan to align existing clusters to this change?
2. How do we track which cluster "architecture" it is in the cluster itself? ARO has a metadata internally but this now makes it 3rd iteration of "Infrastructure changes".
3. How this will play in the long run when you have 2 types of NICs out there and you need to manage loadbalancers memberships? How this is aligned to MachineAPI as they potentially now will have to deal with 3 Azure architectures in the wild:
3.1 Single and multiple NGS (change done long ago)
3.2 Nic names prior this PR
3.3 Nic names after this PR

Basically blind changes like this creates much more complex work for downstream components and it would be good to align this in some way. 

cc @JoelSpeed

**Comment by mtnbikenc** [comment]
I will bring this up with the team.
/hold

**Comment by sdodson** [comment]
@mjudeikis What specifically would you like for the Installer to do with your bug then? Your concern is that we're not coordinating with the cloud infra team to ensure that these remain stable from this point forward?

**Comment by sdodson** [comment]
/lgtm cancel
/approve cancel
Clearing these so someone doesn't inadvertently let this merge by clearing the hold only.

**Comment by JoelSpeed** [comment]
AFAIK this should be safe with respect to MAPI. We only care about the NicName during VM creation, in which case we generate it with https://github.com/openshift/cluster-api-provider-azure/blob/master/pkg/cloud/azure/defaults.go#L128-L130.

When we are updating or deleting VMs, we look up the NIC that is attached to the VM, we don't guess its name based on a pattern or anything.

LGTM from me on this front at least

**Comment by mjudeikis** [comment]
@sdodson My general worry is just about the fact that installer does not have any responsibilities day 2. We already seen multiple cases where now we have 4-5 different flavors of clusters out in the wild, and this would be adding one more.

In general this is a problem which is being pushed down the queue to the last standing person in the line. In most cases its CEE, customer or SRE's. 
At some point we need to stop doing this and start tracking changes in some architecture version per components so at any point in time any component interacting with other component would know which version of resource it need to talk to and how to interact with.

**Comment by sdodson** [comment]
@mjudeikis I think that's a worthwhile gap to identify but I don't think it will be addressed within the scope of a bug or a PR, or even by the Installre team. So need a decision on whether we close this PR and the BZ or not which is made independently of whether you pursue the architectural tracking changes.

**Comment by mjudeikis** [comment]
@sdodson I say merge, but all this is creating quite a tech dept for teams outside AOS... just saying...

**Comment by mtnbikenc** [comment]
@sdodson @mjudeikis , Where do we stand on this PR?  I'm happy to move forward or close this and the bug as WONTFIX.  I'll rebase this in the meantime.

**Comment by sdodson** [comment]
/override ci/prow/e2e-aws-upgrade
This only affects azure
